src/sentry/api/serializers/rest_framework/group_notes.py

```diff
@@ -3,21 +3,62 @@
 from rest_framework import serializers
 
 from .list import ListField
+from sentry.api.fields.actor import ActorField
+
+from sentry.models import User, Team
+
+
+def seperate_actors(actors):
+    users = [actor for actor in actors if actor.type is User]
+    teams = [actor for actor in actors if actor.type is Team]
+
+    return {'users': users, 'teams': teams}
+
+
+def seperate_resolved_actors(actors):
+    users = [actor for actor in actors if isinstance(actor, User)]
+    teams = [actor for actor in actors if isinstance(actor, Team)]
+
+    return {'users': users, 'teams': teams}
 
 
 class NoteSerializer(serializers.Serializer):
     text = serializers.CharField()
-    mentions = ListField(required=False)
+    mentions = ListField(child=ActorField(), required=False)
 
     def validate_mentions(self, attrs, source):
         if source in attrs and 'group' in self.context:
+
             mentions = attrs[source]
+            seperated_actors = seperate_actors(mentions)
+            # Validate that all mentioned users exist and are on the project.
+            users = seperated_actors['users']
+
+            mentioned_user_ids = [user.id for user in users]
+
             project = self.context['group'].project
+
             member_ids = set(
```

Comment:
i know you didn't name this variable, but this should probably be `user_ids` since the current name implies `member.id`

Reply:
good point